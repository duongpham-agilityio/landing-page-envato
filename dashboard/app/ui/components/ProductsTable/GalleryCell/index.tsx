// Libs
import { memo } from 'react';
import { Box, Flex, Td } from '@chakra-ui/react';
import Image from 'next/image';
import isEqual from 'react-fast-compare';

// Constants
import { IMAGES } from '@/lib/constants';

// Utils
import { generatePlaceholder } from '@/lib/utils';

interface GalleryCellProps {
  imageURL?: string;
  name: string;
}

const GalleryCell = ({
  imageURL = IMAGES.SIGN_UP.url,
  name,
}: GalleryCellProps) => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 150, md: 20 }}
  >
    <Flex
      alignItems="center"
      gap="10px"
      minW={180}
      borderRadius="15px"
      w={{ base: 100, '3xl': 150, '5xl': 250 }}
    >
      <Box pos="relative" w={{ base: 50, lg: 100 }} h={{ base: 50, lg: 100 }}>
        <Image
          src={imageURL}
          alt={`Image of ${name}`}
          fill
          sizes="100vw"
          placeholder="blur"
          blurDataURL={generatePlaceholder(40, 40)}
          style={{
            objectFit: 'cover',
            borderRadius: '15px',
          }}
        />
      </Box>
    </Flex>
  </Td>
);

const GalleryCellMemorized = memo(GalleryCell, isEqual);

export default GalleryCellMemorized;
