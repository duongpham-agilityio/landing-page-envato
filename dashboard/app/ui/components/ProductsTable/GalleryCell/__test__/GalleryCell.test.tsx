import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components
import GalleryCell from '..';

const mockProps = {
  name: 'Shoe',
};

describe('GalleryCell component', () => {
  const setup = () =>
    render(<GalleryCell {...mockProps} />, {
      wrapper: Table,
    });

  it('should match with snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
