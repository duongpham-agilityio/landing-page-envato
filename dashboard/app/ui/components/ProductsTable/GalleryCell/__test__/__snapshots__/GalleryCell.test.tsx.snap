// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-oe3t8o"
    >
      <div
        class="css-s2putz"
      >
        <div
          class="css-sdavkg"
        >
          <img
            alt="Image of Shoe"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="/icons/signup.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; border-radius: 15px; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
        </div>
      </div>
    </td>
  </table>
</div>
`;
